proto/v1/instance_service.proto

```diff
@@ -137,6 +145,21 @@ message ListInstancesResponse {
   string next_page_token = 2;
 }
 
+message SearchInstancesRequest {
+  // The parent parameter's value depends on the target resource for the request.
+  // - instances.list(): An empty string. This method doesn't require a resource; it simply returns all instances the user has access to.
+  // - projects.instances.list(): projects/{PROJECT_ID}. This method lists all instances that have databases in the project.
+  string parent = 1;
```

Comment:
Search should be parent-less. Maybe we should use project "filter" for search? Anyway, maybe later.

Reply:
@RainbowDashy TODO(p0ny): Search should be parent-less. Maybe we should use project "filter" for search? Anyway, maybe later.